### DISCUSSION TITLE
Extended timestamp

### DISCUSSION THREAD

**Discussion by isouriadakis** [shogo4405/HaishinKit.kt#129] [Q&A]
Hi,

Thank you for this great library it has been great. However, I am facing an issue, after a while I start getting messages with random streamIds and no header size.

Is this a behaviour you have noticed?

Also is there a way to enable extended Timestamp without modifying the RtmpMuxer to always send video messages as RTMPChunk.ZERO.

How I can disable the automatic addition off timestamp when RTMPChunk.One is chosen. I would like to send a relative timestamp which is smaller than 16777215, not a delta one.

Thanks in advance,
Ioannis

**Comment by shogo4405** [comment]
I've never heard of it. How can I reproduce it?
I have no plan both feature.

**Comment by isouriadakis** [comment]
Hi, any idea on this error? It hapens after streaming for 6-7 hours

`D/FrameTracker(27974): video stats: frames=25, average=40.2, sd=9.095053600721657
D/FrameTracker(27974): video stats: frames=25, average=39.76, sd=9.500652609163225
2
W/Choreographer(27974): Frame time is 0.090808 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
D/FrameTracker(27974): video stats: frames=25, average=40.24, sd=7.306326026122842
D/FrameTracker(27974): video stats: frames=25, average=39.76, sd=8.626841832327749
2
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
5
I/hin_kit_exampl(27974): Waiting for a blocking GC Alloc
I/hin_kit_exampl(27974): Clamp target GC heap from 233MB to 192MB
I/hin_kit_exampl(27974): Alloc concurrent copying GC freed 12429(508KB) AllocSpace objects, 169(5412KB) LOS objects, 3% free, 185MB/192MB, paused 100us total 53.847ms
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 22.396ms
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 22.368ms
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 21.820ms
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 9.210ms
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 5.363ms
3
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
3
I/hin_kit_exampl(27974): Waiting for a blocking GC Alloc
I/hin_kit_exampl(27974): Alloc young concurrent copying GC freed 455(147KB) AllocSpace objects, 7(224KB) LOS objects, 3% free, 186MB/192MB, paused 106us total 13.903ms
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 12.542ms
5
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
3
I/hin_kit_exampl(27974): Waiting for a blocking GC Alloc
I/hin_kit_exampl(27974): Clamp target GC heap from 234MB to 192MB
I/hin_kit_exampl(27974): Alloc concurrent copying GC freed 1144(150KB) AllocSpace objects, 9(288KB) LOS objects, 2% free, 186MB/192MB, paused 94us total 40.234ms
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 10.960ms
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 10.772ms
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 9.583ms
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
D/FrameTracker(27974): video stats: frames=25, average=40.2, sd=11.085125168440817
2
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
2
I/hin_kit_exampl(27974): Waiting for a blocking GC Alloc
I/hin_kit_exampl(27974): Alloc young concurrent copying GC freed 213(118KB) AllocSpace objects, 3(96KB) LOS objects, 2% free, 186MB/192MB, paused 207us total 18.606ms
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 8.230ms
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
I/hin_kit_exampl(27974): WaitForGcToComplete blocked Alloc on HeapTrim for 6.274ms
3
I/hin_kit_exampl(27974): Starting a blocking GC Alloc
2
I/hin_kit_exampl(27974): Waiting for a blocking GC Alloc
I/hin_kit_exampl(27974): Clamp target GC heap from 234MB to 192MB
I/hin_kit_exampl(27974): Alloc concurrent copying GC freed 166(112KB) AllocSpace objects, 3(96KB) LOS objects, 2% free, 186MB/192MB, paused 70us total 41.382ms`
